### DISCUSSION TITLE
Can you see zwave connection range?

### DISCUSSION THREAD

**Discussion by iptvcld** [dgreif/ring#1036] [General]
Can you see **zwave** connection range? Such as the RSSI?

**Comment by dgreif** [comment]
Unfortunately I don't think Ring has that information in the API. Generally if you can see something in the Ring app, it's available for consumption in this library, but zwave devices don't show any info about connection status/range/etc

**Reply by iptvcld** [reply]
Thank you; would you know if this API can show what devices are connected to the base station and what devices are connected to a range extender?

**Reply by dgreif** [reply]
Hmm not sure that info is available either. You could dig around in your data and see if anything points to that type of setup, but not that I'm familiar with

**Reply by iptvcld** [reply]
thank you; I ran the below and i got the below output.  Would you happen to know if the objects under the node indicate a device is connected?

const ZWaveAdapter = devices.find(device => device.data.deviceType === RingDeviceType.ZWaveAdapter)
console.log(ZWaveAdapter.data)

network: {
    lastResetTimestamp: 1661917838499,
    nodes: {
      '6': [Object],
      '7': [Object],
      '8': [Object],
      '9': [Object],
      '10': [Object],
      '11': [Object],
      '12': [Object],
      '13': [Object],
      '14': [Object],
      '15': [Object],
      '16': [Object],
      '17': [Object],
      '18': [Object],
      '19': [Object],
      '21': [Object],
      '24': [Object],
      '25': [Object],
      '26': [Object],
      '27': [Object],
      '28': [Object],
      '29': [Object],
      '31': [Object]
    }

**Reply by dgreif** [reply]
I don't know for certain but that seems logical :)

**Reply by iptvcld** [reply]
thanks; might be getting somewhere with this.. last question, sorry - would you know how i can get the object for each node.  Just trying to see what each node number is.

**Reply by dgreif** [reply]
You can use `console.log(JSON.stringify(ZWaveAdapter.data, null, 2))`
